src/toolchain.mli

```diff
@@ -1,85 +1,45 @@
-(** Toolchain.ml exposes functions that let us
+(** Manages OCaml Platform toolchain
 
-   1. Run initial checks in the environment looking for a reliable toolchain
-   ([Toolchain.init])
+    The toolchain is installed in a private sandbox, managed by VSCode only,
+    where the tools we depend on are installed automatically.
 
-   2. Run a setup that would setup the toolchain provided that basic
-   requirements are met ([Toolchain.run_setup])
+    The sandbox uses Opam, but it does not conflict with users who use Esy as
+    their package managers, since the tools are separated from the project. *)
 
-   3. Helper functions that extract the tools from the setup toolchain. This
-   includes just [ocamllsp] right now, but in future could include others like
-   debuggers, REPLs etc that could be shipped with [vscode-ocaml-platform] plugin
-   itself
+type t
 
-   The separation between [init], [run_setup] and extraction helpers exist so that we
-   can handle missing tools gracefully (ie provide degraded performance, direct
-   user to install missing tools etc). Having a single [Toolchain.make()], for
-   instance, would not make it this flexible. *)
+val is_sandbox_installed : t -> bool Promise.t
 
-type t =
-  | Opam of Opam.t * Opam.Switch.t
-  | Esy of Esy.t * Path.t
-  | Global
-  | Custom of string
+val get_installed_sandbox : unit -> Sandbox.t option Promise.t
 
-val equal : t -> t -> bool
+val setup_toolchain_sandbox :
+     unit
+  -> (unit, [> `Opam_not_available | `Switch_create_failed of string ]) result
```

Comment:
Are we doing something specific for these errors? If not, then there's no need to introduce specific error types.

The old codebase suffered from this a problem: a massive amount of error types that all eventually ended up flushed down as strings. Let's not reintroduce it.